lib/cc/analyzer/engine.rb

```diff
@@ -34,8 +34,12 @@ def run(stdout_io, container_listener)
         )
 
         container.on_output("\0") do |output|
-          unless output_filter.filter?(output)
-            stdout_io.write(output) || container.stop
+          next if output.blank?
+
+          issue_hash = IssueHash.new(output)
```

Comment:
I'm :+1: on @pbrisbin's approach. That was the direction I was trying to move in.

> On Dec 11, 2015, at 14:28, pat brisbin <notifications@github.com> wrote:
> 
> In lib/cc/analyzer/engine.rb:
> 
> > @@ -34,8 +34,12 @@ def run(stdout_io, container_listener)
> >          )
> >  
> >          container.on_output("\0") do |output|
> > -          unless output_filter.filter?(output)
> > -            stdout_io.write(output) || container.stop
> > +          next if output.blank?
> > +
> > +          issue_hash = IssueHash.new(output)
> Mentioned IRL, but for visibility:
> 
> I think we should go with an actual Issue class we can pass around and filter on that responds to methods. Then we #to_json that for the stdout_io.write call.
> 
> —
> Reply to this email directly or view it on GitHub.
>